Automattic/wp-calypso
# Automattic Components: Temporarily shim old calypso select dropdown component

Related to https://github.com/Automattic/wp-calypso/pull/79255
Tracking issue https://github.com/Automattic/martech/issues/2023

## Proposed Changes
- At dale's suggestion, point the old select dropdown component to the new one in @automattic/components, allowing us to transition from the old code to the new in smaller pieces and without risking updates to the old component in the interim.
- Remove old files

## Testing Instructions



* Check out this branch or use calypso.live
* Billing History
  * Navigate to calypso.localhost:3000/purchases/billing-history/{SITE_SLUG}
  * <img width="1493" alt="Screenshot 2023-08-22 at 7 10 27 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/c269e39d-c0bc-460e-b781-71ee2ec03237">
* Domains Management
  * Navigate to calypso.localhost:3000/domains/manage/{SITE_SLUG}
  * <img width="1494" alt="Screenshot 2023-08-22 at 7 13 47 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/859b9b6d-eddd-404c-978b-f3721bc383d0">
*  Themes Showcase
  * Navigate to calypso.localhost:3000/themes/{SITE_SLUG}
  * <img width="1492" alt="Screenshot 2023-08-22 at 7 17 05 PM" src="https://github.com/Automattic/wp-calypso/assets/5414230/b7281205-dc94-4611-94ca-829d8ffc18d7">

* Smoke test other instances of form label to ensure that they are rendered properly

## Pre-merge Checklist
